pkg/providers/instance/instance.go

```diff
@@ -277,11 +287,10 @@ func (p *DefaultProvider) checkODFallback(nodeClaim *karpv1.NodeClaim, instanceT
 	instanceTypeZones := map[string]struct{}{}
 	for _, ltc := range launchTemplateConfigs {
 		for _, override := range ltc.Overrides {
-			if override.InstanceType != nil {
-				instanceTypeZones[*override.InstanceType] = struct{}{}
+			if override.InstanceType != ec2types.InstanceType("") {
```

Comment:
This being empty isn't possible given how we build our overrides in `getOverrides` -- I think we can remove this check

Reply:
+1